packages/fast-usdc-deploy/src/upgrade-evm-solana.core.js

```diff
@@ -59,15 +89,19 @@ export const upgradeEvmSolana = async (
   trace('options', options);
   const {
     agoricToNoble = config.MAINNET.agoricToNoble,
+    customFeeConfigs = config.MAINNET.customFeeConfigs,
     fastUsdcCode = assert.fail('missing bundleID'),
   } = options;
   const fuKit = await fastUsdcKit;
   trace('fastUsdcKit.privateArgs keys:', keys(fuKit.privateArgs));
+  const { brand: usdcBrand } = fuKit.privateArgs.feeConfig.flat;
   const { adminFacet, creatorFacet } = fuKit;
-  const upgraded = await E(adminFacet).upgradeContract(
-    fastUsdcCode.bundleID,
-    fuKit.privateArgs,
-  );
+  const upgraded = await E(adminFacet).upgradeContract(fastUsdcCode.bundleID, {
+    ...fuKit.privateArgs,
+    customFeeConfigs: objectMap(customFeeConfigs, c =>
+      objectMap(c, v => AmountMath.make(usdcBrand, /** @type {bigint} */ (v))),
+    ),
```

Comment:
using `fromExternalConfig` would avoid this sort of gymnastics.

meanwhile, can we use a different name for with and without brands?

Reply:
My brain has difficulty grokking `fromExternalConfig` for whatever reason. Want to huddle and show me?